src/components/QueuedTransactionList/index.tsx

```diff
@@ -0,0 +1,190 @@
+  }
+}
+
+const TxsToConfirmList = (): ReactElement => {
+  const ownedSafes: Record<string, string[]> = useOwnerSafes()
+  const localSafesWithDetails = useLocalSafes()
+  const localSafes: Record<string, string[]> = Object.entries(localSafesWithDetails).reduce(
+    (result, [chain, safes]) => {
+      const safesAddr = safes.map((safe) => safe.address)
+      return { ...result, [chain]: safesAddr }
+    },
+    {},
+  )
+  const userAccount = useSelector(userAccountSelector)
+
+  const [loading, setLoading] = useState<boolean>(true)
+  const [txsAwaitingConfirmation, setTxsAwaitingConfirmation] = useState<TransactionsSummaryPerChain>({})
+  const [displayOwnedSafes, setDisplayOwnedSafes] = useState<boolean>(false)
+  const [isInitialLoad, setIsInitialLoad] = useState<boolean>(true)
+
+  const canDisplayOwnedSafes = displayOwnedSafes && userAccount
+  const safesToTraverse = canDisplayOwnedSafes ? ownedSafes : localSafes
+
+  const title = <h2>Transactions to Sign</h2>
+
+  const handleToggleOwnedSafes = () => {
+    if (userAccount) {
+      setDisplayOwnedSafes((val) => !val)
+      setTxsAwaitingConfirmation({})
+      setIsInitialLoad(true)
+    }
+  }
+
+  // Fetch txs awaiting confirmations after retrieving the owned safes from the LocalStorage
+  useEffect(() => {
+    if (!isInitialLoad || !Object.keys(safesToTraverse || {}).length) {
+      return
+    }
+    const fetchAwaitingConfirmationTxs = async () => {
+      const txs: TransactionsSummaryPerChain = {}
+
+      for (const [chainId, safesPerChain] of Object.entries(safesToTraverse).slice(0, 3)) {
+        txs[chainId] = {}
+        const arrayPromises = safesPerChain.map((safeAddr) => {
+          return getTxsAwaitingConfirmationByChainId(chainId, safeAddr, userAccount, displayOwnedSafes)
+        })
+        const txsByChain = await Promise.all(arrayPromises)
+
+        txsByChain.forEach((summaries, i) => {
+          const safeAddress = safesPerChain[i]
+          txs[chainId][safeAddress] = summaries
+        })
+      }
+      setTxsAwaitingConfirmation(txs)
+      setLoading(false)
+      setIsInitialLoad(false)
+    }
+    fetchAwaitingConfirmationTxs()
+  }, [displayOwnedSafes, userAccount, safesToTraverse])
+
+  if (loading || isInitialLoad) {
+    return (
+      <>
+        {title}
+        <h3>Loading</h3>
+      </>
+    )
+  }
+
+  if (Object.keys(txsAwaitingConfirmation).length === 0) {
+    return (
+      <>
+        {title}
+        <h3>No Transactions</h3>
+      </>
+    )
+  }
+  return (
+    <>
+      {title}
+      <FormControlLabel
+        control={<Switch checked={displayOwnedSafes} onChange={handleToggleOwnedSafes} />}
+        label="Only Owned Safes"
+      />
+      <List component="div">
+        {Object.entries(txsAwaitingConfirmation)?.map(([chainId, transactionsPerSafe]) => {
+          const { shortName } = getChainById(chainId)
+          return Object.entries(transactionsPerSafe).map(([safeAddress, transactions]) => {
```

Comment:
I'd suggest memoising these two as they are very complex data structures. Perhaps the entire return.